snap/local/files/run_nv_hostengine.sh

```diff
@@ -12,3 +12,9 @@ if [ -n "$nv_hostengine_port" ]; then
 fi
 
 exec "$SNAP/usr/bin/nv-hostengine" -n --service-account nvidia-dcgm "${args[@]}"
+
+# Check if dcgm-exporter service is running and restart it
+dcgm_exporter_status=$(snap services dcgm.dcgm-exporter | awk '/dcgm\.dcgm-exporter/ {print $3}')
```

Comment:
does this script run in a context that has access to snapd? And if it does, does it perhaps have access to systemd as well? Because in that case you could simply do
```
systemctl is-active --quiet snap.dcgm.dcgm-exporter.service
```

Reply:
Tried with systemctl, permission denied

Reply:
ok too bad. Then maybe leave a comment about it before someone tries to refactor ;)